core/src/types/tx_data.rs

```diff
@@ -0,0 +1,26 @@
+//! Types that are meant to be serialized and used as the data component of a
+//! Namada transaction.
+
+use borsh::{BorshDeserialize, BorshSerialize};
+
+use super::storage;
+
+/// Represents an arbitrary write to storage at the specified key. This should
+/// be used alongside the test `tx_write.wasm`.
+#[derive(
+    Clone,
+    Debug,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Hash,
+    BorshSerialize,
+    BorshDeserialize,
+)]
+pub struct TxWriteData {
+    /// The storage key to be written to.
+    pub key: storage::Key,
+    /// The bytes to be written.
+    pub value: Vec<u8>,
+}
```

Comment:
I'm not sure where best to put this, it needs to be importable into both the `namada_wasm_for_tests` and `namada_tests` crates. Maybe a `namada_test_utils`  crate like in https://github.com/anoma/namada/pull/893, but have put this here for now.

Reply:
the crate looks like a good idea! In the interim, I think you could put it behind `cfg(feature = "testing")`, but it's not enabled from `wasm_tests` yet

Reply:
I tried this and having to use `namada_core/testing` in the `namada_wasm_for_tests` crate seems to break the wasm build somehow - probably via the pulling in of the `rand` crate. e.g.

```
 > make -C wasm_for_tests/wasm_source tx_write
cargo build --release --target wasm32-unknown-unknown --target-dir 'target' --features tx_write && \
        cp "./target/wasm32-unknown-unknown/release/namada_wasm_for_tests.wasm" ../tx_write.wasm
   Compiling num-traits v0.2.15
   Compiling tempfile v3.3.0
   Compiling wait-timeout v0.2.0
error[E0433]: failed to resolve: use of undeclared crate or module `imp`
  --> /Users/main/.cargo/registry/src/github.com-1ecc6299db9ec823/wait-timeout-0.2.0/src/lib.rs:66:9
   |
66 |         imp::wait_timeout(self, dur)
   |         ^^^ use of undeclared crate or module `imp`

For more information about this error, try `rustc --explain E0433`.
```

I can try out putting this in a `namada_test_utils` crate now since that should ultimately be less hacky.

Reply:
I've put this in a `namada_test_utils` crate now